sycl/include/sycl/ext/oneapi/matrix/static-query-use.hpp

```diff
@@ -482,6 +482,379 @@ struct matrix_params<
   template <typename Group>
   using joint_matrix_d = joint_matrix<Group, Td, use::accumulator, M, N>;
 };
+
+//////////////////////////////////////////////
+/// AMD Matrix Cores - GFX90A architecture ///
+//////////////////////////////////////////////
+
+template <typename Ta, typename Tc>
+constexpr bool is_combination_valid_amd_gfx90a(size_t sM, size_t sN,
+                                               size_t sK) {
+  if ((std::is_same_v<Ta, half> && std::is_same_v<Tc, float> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, unsigned short> && std::is_same_v<Tc, float> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, double> && std::is_same_v<Tc, double> &&
+       (sM == 16 && sN == 16 && sK == 4)))
+    return true;
+  else
+    return false;
+}
+
+template <typename Ta, typename Tc>
+constexpr bool are_types_valid_amd_gfx90a() {
+  if ((std::is_same_v<Ta, half> && std::is_same_v<Tc, float>) ||
+      (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>) ||
+      (std::is_same_v<Ta, unsigned short> && std::is_same_v<Tc, float>) ||
+      (std::is_same_v<Ta, double> && std::is_same_v<Tc, double>))
+    return true;
+  else
+    return false;
+}
+
+// Default-values query:
+// Specialization for when only types are given, need to query only sizes
+template <typename Ta, typename Tb, typename Tc, typename Td>
+struct matrix_params<
+    architecture::amd_gpu_gfx90a, Ta, Tb, Tc, Td, 0, 0, 0,
+    typename std::enable_if<(
+        !std::is_same_v<Ta, void> && !std::is_same_v<Tb, void> &&
+        !std::is_same_v<Tc, void> && !std::is_same_v<Td, void> &&
+        std::is_same_v<Ta, Tb> && std::is_same_v<Tc, Td>)>::type> {
+
+  static_assert(
+      (are_types_valid_amd_gfx90a<Ta, Tc>()),
+      "Invalid types for AMD gfx90a, supported types are half, float, "
+      "int8_t, int32_t, double and bf16 (Note that unsigned short"
+      "should be used in the DPC++ code to implement bf16) ");
+
```

Comment:
I have used `bfloat16` in DPC++  code for instance in  [joint_matrix_hip_half_gfx90a.cpp](https://github.com/mmoadeli/llvm/blob/05d2e9d2d883244ac68afb9852f20261727d2f4c/sycl/test-e2e/Matrix/joint_matrix_hip_half_gfx90a.cpp) test.

Reply:
Changed to `bfloat16`